src/infrastructure/amqp/AbstractConsumer.ts

```diff
@@ -0,0 +1,168 @@
+import type { Either } from '@lokalise/node-core'
+import type { Channel, Connection, Message } from 'amqplib'
+import type { ZodSchema } from 'zod'
+
+import type { Dependencies } from '../diConfig'
+import { globalLogger, resolveGlobalErrorLogObject } from '../errors/globalErrorHandler'
+
+import type { ConsumerErrorResolver } from './ConsumerErrorResolver'
+import type { CommonMessage } from './MessageTypes'
+import { AmqpMessageInvalidFormat, AmqpValidationError } from './amqpErrors'
+import { deserializeMessage } from './messageDeserializer'
+
+export interface Consumer {
+  consume(): void
+  close(): Promise<void>
+}
+
+const ABORT_EARLY_EITHER: Either<'abort', never> = {
+  error: 'abort',
+}
+
+export type ConsumerParams<MessagePayloadType extends CommonMessage> = {
+  queueName: string
+  messageSchema: ZodSchema<MessagePayloadType>
+}
+
+export abstract class AbstractConsumer<MessagePayloadType extends CommonMessage>
+  implements Consumer
+{
+  protected readonly queueName: string
+  protected readonly connection: Connection
+  // @ts-ignore
+  protected channel: Channel
+  protected readonly errorHandler: ConsumerErrorResolver
```

Comment:
:green_circle: Would be nice to have the same name of the field like the its type: `errorResolver`.